protos/feast/core/Transformation.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+package feast.core;
+
+option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
+option java_outer_classname = "FeatureTransformationProto";
+option java_package = "feast.proto.core";
+
+import "google/protobuf/duration.proto";
+
+// Serialized representation of python function.
+message UserDefinedFunctionV2 {
+    // The function name
+    string name = 1;
+
+    // The python-syntax function body (serialized by dill)
+    bytes body = 2;
+
+    // The string representation of the udf
+    string body_text = 3;
+}
+
+// A feature transformation executed as a user-defined function
+message FeatureTransformationV2 {
+  // Note this Transformation starts at 5 for backwards compatibility
+  oneof transformation {
+    UserDefinedFunctionV2 user_defined_function = 5;
+    OnDemandSubstraitTransformationV2 on_demand_substrait_transformation = 6;
+  }
```

Comment:
I think it's a good time to rethink the naming here. My first suggestion was to rename the field (not message type) to `on_demand_pandas_transformation` instead of `user_defined_function`. But on second thought, since we are also aiming to reuse this in `StreamFeatureViews`, I think protos should no longer be called `OnDemand..`. What do you think? I'm thinking of something like this:

```
UserDefinedFunctionV2 pandas_transformation = 1;
SubstraitTransformationV2 substrait_transformation = 2;

Reply:
I was planning on doing that in a follow up PR to not add too much complexity here.